Homework2: Yuliya Golovach

YuliyaGolovach opened the pull request.

YuliyaGolovach committed:
Fix of a code style warnings.

YuliyaGolovach committed:
Adjust line lengths to conform code style.

YuliyaGolovach committed:
Some whitespaces added, import order fixed.

YuliyaGolovach committed:
New line at the end of file added, whitespaces added.

YuliyaGolovach committed:
Homework 2 and checkio tasks.

YuliyaGolovach committed:
Change some multiline comments.

YuliyaGolovach committed:
Removed too long line comment to pass CI-checks.

YuliyaGolovach committed:
String tasks.

YuliyaGolovach committed:
Added a new line at the end of file to pass CI-tests.

YuliyaGolovach committed:
Remove unnecessary files from homework 2

YuliyaGolovach committed:
Add one task from Hackerrank

YuliyaGolovach committed:
Fix flake8 errors

Fix docstring and fix import style.

YuliyaGolovach committed:
Fix flake8 errors

Fix docstring style.

belovmd commented on src/hw_2/castle_on_the_grid.py:
```diff
@@ -0,0 +1,142 @@
+
+
+def print_moves(moves, n):
+    # print(moves)
```
debug comments ^)

belovmd commented on src/hw_2/castle_on_the_grid.py:
```diff
@@ -0,0 +1,142 @@
+        cell = q.popleft()
+
+        # north
+        x = cell.x - 1
```
*you don't need to fix it* we can make func move that will take tuple `(dicretion_x, direction_y)` and handle `allowed[x+direction_x][y + direction_y]` and it would be no need to make 4 whiles

YuliyaGolovach replied:
Thanks! Next time will use it (but not in XO-game, too late).

belovmd commented on src/hw_2/castle_on_the_grid.py:
```diff
@@ -0,0 +1,142 @@
+    assert(minimum_moves(grid2, 1, 4, 0, 0)) == 5, 'Test 3'
+
+    # example:
+    # print(minimum_moves(grid2, 1, 4, 0, 0))
```
debug comments ^)

belovmd commented on src/hw_2/checkio_easy_unpack.py:
```diff
@@ -0,0 +1,27 @@
+    first = elements[0]
+    second = elements[2]
+    third = elements[-2]
+    return first, second, third
```
it will be ok if it would be single-line solution. no need to fix it

belovmd commented on src/hw_2/checkio_stressful_subject.py:
```diff
@@ -0,0 +1,69 @@
+
+    # import itertools
+    #
+    # def is_stressful(subj):
```
debug comments

belovmd commented on src/hw_2/checkio_stressful_subject.py:
```diff
@@ -0,0 +1,69 @@
+subject line means that
+ - all letters are in uppercase,
+ - and/or ends by at least 3 exclamation marks,
+  - and/or contains at least one of the following “red” words:
```
misprint - strange indentation

belovmd reviewed (changes requested):
Thank you for your work. 
Please address next paragraphs from review notes [Review Notes](https://github.com/belovmd/it-academy-python-spring/blob/master/homework_review/Homework2.md):
- 13
- 15

YuliyaGolovach committed:
Make fixes after Code Review

Add __init__.py file.
Rename files.
Address task 5 note.
Add one more task from Hackerrank.
Remove redundant comments.

YuliyaGolovach committed:
Add Anagrams task

Add Anagrams task from Hackerrank, delete elementary task from Checkio.

belovmd reviewed (approved):
10
